### ISSUE TITLE
Add block timestamp to contract events

### ISSUE THREAD

**Issue by janayama** [ConsenSys/eventeum#24]
Hey there! 

Currently in our use case we would like to store the block timestamp of a contract event. Since we do not have that block timestamp when is creating the event [in this factory](https://github.com/ConsenSys/eventeum/blob/master/core/src/main/java/net/consensys/eventeum/chain/service/factory/DefaultContractEventDetailsFactory.java#L43).

We decided to ask for the block info when having the block number here: https://github.com/ConsenSys/eventeum/blob/master/core/src/main/java/net/consensys/eventeum/chain/service/Web3jService.java#L100

That means that every time that we receive a contract event in that observable we call the function ` this.web3j.ethGetBlockByNumber` to get the block information so we will receive the timestamp.

We wanted to share with you @craigwilliams84 this solution and discuss if you think is a good approach or you have other thing in mind.

Best regards and thanks in advance for this great library!

**Comment by craigwilliams84** [comment]
Hey, thanks for getting in touch, and proposing this solution.

This would work for all cases besides Infura.  This is because Infura load balance across multiple nodes and don't yet support sticky sessions.  This leaves a situation where a get by block number can fail because the block has not yet been propagated to the node that handled the request (due to network latency etc).

You aren't the first person to request this feature so I might try and work on an implementation soon.  I've got an idea in mind; having a short lived in memory cache of blocks populated my a BlockListener (the timestamp information is included in new block notifications), that can be queried when a new contract event is received by Eventeum.

**Comment by janayama** [comment]
Yup! Actually that was the first thing I've checked, if eventeum was storing that block info in mongo to get it from there, but didn't find.

I understand what you said, but could it be, since you have two observables, one for contract events and other for blocks, that you receive first the contract event without having yet the in-memory block structure updated? I though about that solution but I discarded cause maybe will be some race conditions there... not sure afaik.
